aws/rust-runtime/aws-config/src/meta/credentials/chain.rs

```diff
@@ -95,6 +98,35 @@ impl CredentialsProviderChain {
             "no providers in chain provided credentials",
         ))
     }
+
+    async fn credentials_with_timeout(
+        &self,
+        sleeper: Arc<dyn AsyncSleep>,
+        timeout: Duration,
+    ) -> provider::Result {
+        for (name, provider) in &self.providers {
+            let timeout_per_provider = timeout.div_f64(self.providers.len() as f64);
```

Comment:
This cannot be a simple division as each provider has a different idea of how long timeout should be. This implies that we should consider what it means for a credentials provider to time out in a broader scope.

Reply:
At least preserved today's behavior with 4592ca5, but a set of APIs `provide_credentials` and `provide_credentials_with_timeout` needs to be reconsidered.